src/components/Carousel.js

```diff
@@ -280,16 +282,39 @@ export default class Carousel extends Component {
    * throttled to improve performance
    * @type {Function}
    */
-   onResize = throttle(() => {
-     const arrowLeftWidth = this.arrowLeftNode && this.arrowLeftNode.offsetWidth;
-     const arrowRightWidth = this.arrowRightNode && this.arrowRightNode.offsetWidth;
-     const width = this.node.offsetWidth - (arrowLeftWidth || 0) - (arrowRightWidth || 0);
+  onResize = throttle(() => {
+    const arrowLeftWidth = this.arrowLeftNode && this.arrowLeftNode.offsetWidth;
+    const arrowRightWidth = this.arrowRightNode && this.arrowRightNode.offsetWidth;
+    const width = this.node.offsetWidth - (arrowLeftWidth || 0) - (arrowRightWidth || 0);
 
-     this.setState({
-       carouselWidth: width,
-       windowWidth: window.innerWidth,
-     });
-   }, config.resizeEventListenerThrottle);
+    this.setState({
+      carouselWidth: width,
+      windowWidth: window.innerWidth,
+    });
+
+    this.getProp('dots') && this.setNumDots();
+  }, config.resizeEventListenerThrottle);
+
+  setNumDots = () => {
+    const numSlides = this.getChildren().length;
+    const slidesPerPage = this.getProp('slidesPerPage');
+    const slidesPerScroll = this.getProp('slidesPerScroll');
+    let numDots = Math.ceil((numSlides - slidesPerPage) / slidesPerScroll);
+
+    if (!numDots) {
+      numDots = 1;
+    }
+
+    if (numSlides % slidesPerPage || slidesPerPage === 1) {
+      numDots++;
+    }
+
+    if (numDots !== this.state.numDots) {
+      this.setState({ value: 0 });
```

Comment:
actually, this is supposed to move the slider to 0 when a configured breakpoint is hit, and this isn't working. couldn't get `this.changeSlide(0)` to work with the unit tests though.